kong/runloop/plugin_servers/pb_rpc.lua

```diff
@@ -409,6 +409,10 @@ function Rpc:handle_event(plugin_name, conf, phase)
       self.reset_instance(plugin_name, conf)
       kong.log.warn(err)
       return self:handle_event(plugin_name, conf, phase)
+
+    elseif err == "timeout" then
+      kong.log.err("pluginserver socket timeout")
+      kong.response.error(500)
     end
 
     kong.log.err(err)
```

Comment:
Instead of catching the timeout specifically, would it be better to throw a 500 for _all_ unhandled errors?

```suggestion
    kong.log.err(err or "unknown plugin server error")
    return kong.response.error(500)
```

If a new error string is introduced or the `"timeout"` error string is changed without our knowledge, we'll still abort and return 500.

Reply:
Yes, if other kinds of errors are also unexpected, we should handle them as well instead of skipping the plugin.